src/ensembl/src/genome/genomeActions.ts

```diff
@@ -21,12 +28,30 @@ export const fetchGenomeInfoAsyncActions = createAsyncAction(
 
 export const fetchGenomeData: ActionCreator<
   ThunkAction<void, any, null, Action<string>>
-> = (genomeId: string) => async (dispatch) => {
+> = (genomeId: string) => async (dispatch, getState: () => RootState) => {
   await Promise.all([
     dispatch(fetchGenomeInfo(genomeId)),
     dispatch(fetchGenomeTrackCategories(genomeId))
   ]);
 
+  const state = getState();
+  const committedSpecies: CommittedItem[] = getEnabledCommittedSpecies(state);
+  const genomeInfo: GenomeInfo | null = getGenomeInfoById(state, genomeId);
+  if (
+    genomeInfo &&
+    !find(
+      committedSpecies,
+      (species: CommittedItem) => species.genome_id === genomeId
+    )
+  ) {
+    const newCommittedSpecies = [
+      ...committedSpecies,
+      { ...omit(genomeInfo, ['example_objects']), isEnabled: true }
+    ];
```

Comment:
The added code in green does not really belong to a function whose name is `fetchGenomeData`. In the spirit of keeping functions small, can we move this code to its own thunk function (`addSpeciesFromGenomeId`?)?

Reply:
That's done